pipenv/utils.py

```diff
@@ -83,23 +83,26 @@ def convert_deps_to_pip(deps, r=True):
         if 'version' in deps[dep]:
             version = deps[dep]['version']
 
-        # Support for git.
-        # TODO: support SVN and others.
-        if 'git' in deps[dep]:
-            extra = 'git+{0}'.format(deps[dep]['git'])
-
-            # Support for @refs.
-            if 'ref' in deps[dep]:
-                extra += '@{0}'.format(deps[dep]['ref'])
-
-            extra += '#egg={0}'.format(dep)
-
-            # Support for editable.
-            if 'editable' in deps[dep]:
-                # Support for --egg.
-                dep = '-e '
-            else:
-                dep = ''
+        # Support for version control
+        maybe_vcs = [vcs for vcs in ('git', 'svn', 'hg', 'bzr') if vcs in deps[dep]]
+        vcs = maybe_vcs[0] if maybe_vcs else None
+
+        if vcs:
+            extra = '{0}+{1}'.format(vcs, deps[dep][vcs])
+
+            if extra:
```

Comment:
Is there a reason for adding this conditional? Given `extra = '{0}+{1}'.format(vcs, deps[dep][vcs])`, when can `extra` be a non-truthy value?